internal/decoder/func.go

```diff
@@ -0,0 +1,108 @@
+package decoder
+
+import (
+	"bytes"
+	"unsafe"
+
+	"github.com/goccy/go-json/internal/errors"
+	"github.com/goccy/go-json/internal/runtime"
+)
+
+type funcDecoder struct {
+	typ        *runtime.Type
+	structName string
+	fieldName  string
+}
+
+func newFuncDecoder(typ *runtime.Type, structName, fieldName string) *funcDecoder {
+	fnDecoder := &funcDecoder{typ, structName, fieldName}
+	return fnDecoder
+}
+
+func (d *funcDecoder) DecodeStream(s *Stream, depth int64, p unsafe.Pointer) error {
+	s.skipWhiteSpace()
+	start := s.cursor
+	if err := s.skipValue(depth); err != nil {
+		return err
+	}
+	src := s.buf[start:s.cursor]
+	if len(src) > 0 {
+		switch src[0] {
```

Comment:
It seems better to be able to make an error even if the values ​​are different (e.g. `true` or` false` or other characters).

Reply:
I've added the cases for `true` and `false`.

The stdjson returns an error `invalid character 'a' looking for beginning of value`. This could be added to `internal/errors/errors.go` for other unexpected characters.